net/nginx-util/src/px5g-openssl.hpp

```diff
@@ -88,85 +89,58 @@ auto gen_eckey(const int curve) -> EVP_PKEY_ptr
     }
 
     EC_GROUP_set_asn1_flag(group, OPENSSL_EC_NAMED_CURVE);
-
     EC_GROUP_set_point_conversion_form(group, POINT_CONVERSION_UNCOMPRESSED);
 
-    auto* eckey = EC_KEY_new();
+    EVP_PKEY_CTX* ctx = EVP_PKEY_CTX_new_id(EVP_PKEY_EC, nullptr);
 
-    if (eckey != nullptr) {
-        if ((EC_KEY_set_group(eckey, group) == 0) || (EC_KEY_generate_key(eckey) == 0)) {
-            EC_KEY_free(eckey);
-            eckey = nullptr;
-        }
+    if (!EVP_PKEY_paramgen_init(ctx)) {
+        throw std::runtime_error("Could not init paramgen");
     }
 
-    EC_GROUP_free(group);
+    EVP_PKEY_CTX_set_ec_paramgen_curve_nid(ctx, curve);
 
-    if (eckey == nullptr) {
-        std::string errmsg{"gen_eckey error: cannot build key with curve id "};
-        errmsg += std::to_string(curve) + "\n";
+    EVP_PKEY* params = nullptr;
+    EVP_PKEY_paramgen(ctx, &params);
+
+    EVP_PKEY_CTX* key_gen_ctx = EVP_PKEY_CTX_new(params, nullptr);
+
+    if (EVP_PKEY_keygen_init(key_gen_ctx) <= 0) {
```

Comment:
Missing EVP_PKEY_CTX_new on the error path?

Reply:
You're right, good catch. I'll make the update and check if I'm forgetting to free contexts else where. Thanks.